demo/src/screens/foundationScreens/ColorsScreen.js

```diff
@@ -281,11 +291,99 @@ class ColorsScreen extends Component {
     );
   }
 
+  /** Color Palette */
+
+  showColorPicker = () => {
+    this.setState({showPicker: true});
+  };
+
+  onValueChange = (color) => {
+    this.setState({currentColor: color});
+  };
+
+  onSubmit = (color) => {
+    this.setState({currentColor: color});
+  };
+
+  onDismiss = () => {
+    this.setState({showPicker: false});
+  };
+
+  setDefaultOptions = () => {
+    const designKitsOptions = {adjustLightness: false, adjustSaturation: false, addDarkestTints: true, avoidReverseOnDark: true};
+    const defaultOptions = {adjustLightness: true, adjustSaturation: true, addDarkestTints: false, avoidReverseOnDark: false};
+    if (this.state.isDefaultOptions) {
+      this.setState({...designKitsOptions, isDefaultOptions: false});
+    } else {
+      this.setState({...defaultOptions, isDefaultOptions: true});
+    }
+  };
+
+  renderColorPicker = () => {
+    const {showPicker, currentColor} = this.state;
+    return (
+      <ColorPickerDialog
+        visible={showPicker}
+        initialColor={currentColor}
+        key={currentColor}
+        onDismiss={this.onDismiss}
+        onSubmit={this.onSubmit}
+      />
+    );
+  };
+
+  renderOptions = () => {
+    return (
+      <View padding-20>
+        {renderBooleanOption.call(this, 'adjustLightness', 'adjustLightness')}
+        {renderBooleanOption.call(this, 'adjustSaturation', 'adjustSaturation')}
+        {renderBooleanOption.call(this, 'addDarkestTints', 'addDarkestTints')}
+        {renderBooleanOption.call(this, 'avoidReverseOnDark', 'avoidReverseOnDark')}
+        <Button label={this.state.isDefaultOptions ? 'Reset example' : 'Set defaults'} onPress={this.setDefaultOptions}/>
+      </View>
+    );
+  };
+
+  renderColorPalette = () => {
+    const {currentColor, adjustLightness, adjustSaturation, addDarkestTints, avoidReverseOnDark} = this.state;
+    const paletteOptions = {adjustLightness, adjustSaturation, addDarkestTints, avoidReverseOnDark};
+    const palette = Colors.generateColorPalette(currentColor, paletteOptions);
+    return (
+      <View margin-12 br40 style={{borderWidth: 1}}>
+        {this.renderOptions()}
+        <View center row>
+          <TouchableOpacity
+            marginH-10 
+            style={[styles.colorBox, {backgroundColor: currentColor}]} 
+            onPress={this.showColorPicker}
+          />
+          <ColorPalette
+            colors={palette}
+            value={currentColor}
+            swatchStyle={styles.swatchStyle}
+            containerStyle={{marginLeft: -10}}
+            onValueChange={this.onValueChange}
+          />
+        </View>
+      </View>
+    );
+  };
+
+  renderColorPaletteExample = () => {
+    return (
+      <>
+        {this.renderColorPalette()}
+        {this.renderColorPicker()}
+      </>
+    );
+  };
+
   render() {
     return (
       <>
         {this.renderSearch()}
         <ScrollView ref={this.scrollViewRef}>
+          {/* {this.renderColorPaletteExample()} */}
```

Comment:
Not sure why it's hidden, wouldn't UX will want to test it?
BTW, I did not test it.
We can go over it together if you like

Reply:
Moved to the bottom of the screen